### PR TITLE
Add pull request configuration for buildkite pr bot

### PR DESCRIPTION
**Pull request by mrodm**
Add configuration to configure Buildkite PR bot in order to to enable GitHub comments to trigger builds.

### CODE REVIEW

File: .buildkite/pull-requests.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+   "jobs": [
+     {
+        "enabled": true,
+        "pipelineSlug": "elastic-package",
+        "allow_org_users": true,
+        "allowed_repo_permissions": ["admin", "write"],
+        "allowed_list": ["dependabot[bot]", "mergify[bot]"],
```
Comment (mrodm) [inline .buildkite/pull-requests.json:8]: Not sure if it is needed. Should we remove them and see how it behaves ?

File: .buildkite/pull-requests.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        "set_commit_status": true,
+        "build_on_commit": true,
+        "build_on_comment": true,
+        "trigger_comment_regex": "^(?:(?:buildkite\\W+)?(?:build|test)\\W+(?:this|it))",
+        "always_trigger_comment_regex": "^(?:(?:buildkite\\W+)?(?:build|test)\\W+(?:this|it))",
```
Comment (jsoriano) [inline .buildkite/pull-requests.json:13]: Is there any difference between these two triggers?

**Review by jsoriano — APPROVED** [review]

File: .buildkite/pull-requests.json (hunk 1 above)
Comment (alexsapran) [reply .buildkite/pull-requests.json:8]: Did you copy this from somewhere?
I am not sure if we are using any of the above bots, do we?

Comment (mrodm) [reply .buildkite/pull-requests.json:8]: In elastic-package there are PRs created and merged by these two bots. For instance this one
https://github.com/elastic/elastic-package/commit/ed8b57f1bc760f05f7ea1cd6b4f83f9fdfde19ed

But not sure if it is needed to explicitly to add those users in this list.

Comment (cachedout) [reply .buildkite/pull-requests.json:8]: It's not totally clear to me what this setting does exactly, so please ignore me if this doesn't apply, but be advised that giving bots the right to merge things without human review might put you in a position wherein you can't [adhere to SLSA guidelines](https://slsa.dev/spec/v0.1/requirements#two-person-reviewed). Just FYI. :)

Comment (mrodm) [reply .buildkite/pull-requests.json:8]: > It's not totally clear to me what this setting does exactly, so please ignore me if this doesn't apply, but be advised that giving bots the right to merge things without human review might put you in a position wherein you can't [adhere to SLSA guidelines](https://slsa.dev/spec/v0.1/requirements#two-person-reviewed). Just FYI. :)

@cachedout is it expected that we follow those guidelines ? currently, there are some automations to merge PRs created by dependabot using mergify.
cc @elastic/ecosystem

**Pull request merged**